add rusticl to the site

norablackcat opened the pull request:
This splits Clover and Rusticl into two different sections and handles them independently 

Parsing based on this merge req https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/23055

closes: https://github.com/MightyCreak/mesamatrix/issues/269

norablackcat commented:
Everything should be ready now baring any changes from reviewers on mesa

MightyCreak commented on src/Leaderboard/Leaderboard.php:
```diff
@@ -56,7 +56,9 @@ public function load(\SimpleXMLElement $xmlApi)
-                $a->getName() === Constants::OPENCL_NAME
+                $a->getName() === Constants::CLOVER_OPENCL_NAME ||
+                $a->getName() === Constants::RUSTICL_OPENCL_NAME
+
```
Just an extra newline that can be removed

MightyCreak reviewed (changes requested):
Just a tiny change, but LGTM, thanks!

norablackcat commented:
Upstream has been merged :crab:

illwieckz commented:
I tested this, that looks good! 🙂️

Why not listing RustiCL before Clover? I know RustiCl is still experimental, but Clover is not less experimental, just experimental since more time, and is less complete.

zboszor commented:
> Why not listing RustiCL before Clover? I know RustiCl is still experimental, but Clover is not less experimental, just experimental since more time, and is less complete.

I think it's not really interesting. But it fits alphabetical order: Clover < Rusticl.

norablackcat commented:
need to update this to add r600 as it's been merged so do not merge yet. 



> > Why not listing RustiCL before Clover? I know RustiCl is still experimental, but Clover is not less experimental, just experimental since more time, and is less complete.
> 
> I think it's not really interesting. But it fits alphabetical order: Clover < Rusticl.

Yeah that's the order I did it in initially although it's probably better to put rusticl above because it supports more and is actually maintained

norablackcat commented:
Changed to display rusticl first and added the possible drivers rusticl will support in the future.

Need to double check with Karol to make sure those drivers are correct

norablackcat commented:
It was decided that we are not going to list unsupported drivers and just update the list later when the need arises for some wip driver support

this should be ready for merge

MightyCreak commented:
Hmmm there are some issues with the tests as well, I'm taking a look at it right now.

MightyCreak reviewed (approved):
Pushed some lint and tests fixes, but it's all good now!
